test/helpers/run-wrangler.js

```diff
@@ -62,70 +62,58 @@ async function runLongLivedWrangler (
-    resolveReadyPromise = resolve
-    rejectReadyPromise = reject
-  })
-
-  const wranglerProcess = fork(wranglerEntryPath, command, {
-    stdio: ['ignore', /* stdout */ 'pipe', /* stderr */ 'pipe', 'ipc'],
-    cwd,
-    env: { ...process.env, ...env, PWD: cwd }
-  }).on('message', (message) => {
-    if (settledReadyPromise) return
-    settledReadyPromise = true
-    clearTimeout(timeoutHandle)
-    resolveReadyPromise(JSON.parse(message.toString()))
-  })
+  return new Promise((resolve, reject) => {
+    const wranglerProcess = fork(wranglerEntryPath, command, {
+      stdio: ['ignore', /* stdout */ 'pipe', /* stderr */ 'pipe', 'ipc'],
+      cwd,
+      env: { ...process.env, ...env, PWD: cwd }
+    }).on('message', (messageJSON) => {
+      clearTimeout(timeoutHandle)
+      /** @type {{ ip: string, port: number }} */
+      const message = JSON.parse(messageJSON.toString())
+      resolve({ ...message, stop, getOutput, clearOutput })
+    })
 
-  const chunks = []
-  wranglerProcess.stdout?.on('data', (chunk) => {
-    chunks.push(chunk)
-  })
-  wranglerProcess.stderr?.on('data', (chunk) => {
-    chunks.push(chunk)
-  })
-  const getOutput = () => Buffer.concat(chunks).toString()
-  const clearOutput = () => (chunks.length = 0)
+    /** @type {Buffer[]} */
+    const chunks = []
+    wranglerProcess.stdout?.on('data', (chunk) => {
+      chunks.push(chunk)
+    })
+    wranglerProcess.stderr?.on('data', (chunk) => {
+      chunks.push(chunk)
+    })
+    const getOutput = () => Buffer.concat(chunks).toString()
+    const clearOutput = () => (chunks.length = 0)
 
-  const timeoutHandle = setTimeout(() => {
-    if (settledReadyPromise) return
-    settledReadyPromise = true
-    const separator = '='.repeat(80)
-    const message = [
-      'Timed out starting long-lived Wrangler:',
-      separator,
-      getOutput(),
-      separator
-    ].join('\n')
-    rejectReadyPromise(new Error(message))
-  }, 50_000)
+    const timeoutHandle = setTimeout(() => {
+      const separator = '='.repeat(80)
+      const message = [
+        'Timed out starting long-lived Wrangler:',
+        separator,
+        getOutput(),
+        separator
+      ].join('\n')
+      reject(new Error(message))
+    }, 50_000)
 
-  async function stop () {
-    return new Promise((resolve) => {
-      assert(
-        wranglerProcess.pid,
-                `Command "${command.join(' ')}" had no process id`
-      )
-      treeKill(wranglerProcess.pid, (e) => {
-        if (e) {
-          console.error(
-            'Failed to kill command: ' + command.join(' '),
-            wranglerProcess.pid,
-            e
-          )
-        }
-        resolve()
+    /** @type {WranglerProcessInfo['stop']} */
+    async function stop () {
+      return new Promise((resolve) => {
+        assert(
+          wranglerProcess.pid,
+                  `Command "${command.join(' ')}" had no process id`
+        )
+        treeKill(wranglerProcess.pid, (e) => {
```

Comment:
**suggestion:**
I've tested this locally, and `SIGKILL` was the only reliable way to fully terminate the process after the test was completed. The default `SIGTERM` (standard termination signal) left the process running in the background, likely because the process did not handle the signal as expected.

Since `SIGKILL` forcefully terminates the process without allowing for cleanup, it's a more immediate solution that ensures processes don't linger in the background. This is especially useful for our test environment. While this may bypass graceful shutdown procedures, it prevents the accumulation of stuck processes when running tests multiple times, ensuring a clean state in development.

```suggestion
        treeKill(wranglerProcess.pid, 'SIGKILL', (e) => {
```